subsys/bluetooth/common/bt_str.c

```diff
@@ -61,3 +61,88 @@ const char *bt_uuid_str(const struct bt_uuid *uuid)
 
 	return str;
 }
+
+const char *bt_hci_err_to_str(uint8_t hci_err)
+{
+	#define _HCI_ERR(err) [err] = #err
```

Comment:
Just FYI: The C standard disallows for starting variables and macros with underscore: https://github.com/zephyrproject-rtos/zephyr/pull/73039#discussion_r1616678017